What: Resolve #788 and clarify end users, adopters, & vendors (#864)
cncf/toc: FAQ.md

The CNCF provides a variety of services that are accessible by maintainers via the ServiceDesk: https://github.com/cncf/servicedesk

> ## What is the definition of an adopter?
> Adopters of a CNCF project are any organization that successfully leverages that project in the manner it was intended or repackages it as a core component of a service offering. To increase understanding of how the project is adopted, we distinguish a direct adopter from a transitive adopter; i.e. a Service Provider using Kubernetes as the underlying container orchestrator for their Knative offering, in this scenario the Service Provider is a direct adopter of Kubernetes and their Knative users are direct adopters of Knative but transitive adopters of Kubernetes.

Reviewer: ```suggestion
## What are direct and transitive Adopters?
We distinguish between _direct_ and _transitive_ adopters. 
_Direct_ Adopters of a CNCF project are any organization which successfully uses said project in the manner it was intended (End User) and/or provides it as part of a service (Service Provider). 
_Transitive_ adopters consume projects, commonly from Service Providers.

As an example, an organization using Kubernetes as the underlying container orchestrator for their Knative offering would be a _direct_ Adopter of both Kubernetes and Knative. Their customers would be _direct_ Adopters of Knative and _transitive_ adopters of Kubernetes.
```

Reviewer 2: Wait, I was with you until the example.  How are those customers direct adopters of Knative.  Shouldn't they be transitive adopters of both?

Reviewer 2: Also ... what is the language "in the manner in which it was intended" meant to convey?  For example, if a bus company use Kubernetes to update mileage software on their vehicle fleet, do they not count as end-users because they hacked it too much?  What are we excluding here?

Author: Provided additional clarity in line 57, please take a look

Reviewer 2: I'm still not understanding the example; why is the non-SP org a direct adopter of Knative in this example?   (given that I'm likely to directly help with the eventual Knative Graduated application, this isn't an idle question, I'm asking because it will matter).

For "in the manner in which it was intended" I'm guessing that's there because we want to exclude anyone vendoring individual libraries from the definition of "adopter", as in someone who copied the GH interaction code of Prow, but nothing else, would not be an "adopter"?

The CNCF provides a variety of services that are accessible by maintainers via the ServiceDesk: https://github.com/cncf/servicedesk

## What is the definition of an adopter?
Adopters of a CNCF project are any organization that successfully leverages that project in the manner it was intended or repackages it as a core component of a service offering.

> The TOC’s intent of identifying adopters is to better understand the operational or production-level use of a given project (which can also be a specification) by its operators or users. We do this to ascertain the level of maturity the project has reached, its interactions with adopters, and its likelihood to continue growing within and supporting the ecosystem.

Reviewer 3: ```suggestion
The TOC’s intent in identifying adopters is to better understand the operational or production-level use of a given project (which can also be a specification) by its operators or users. We do this to ascertain the level of maturity the project has reached, its interactions with adopters, and its likelihood to continue growing within and supporting the ecosystem.
```

> [Cloud native](https://github.com/cncf/toc/blob/main/DEFINITION.md) project adopters may be any one of the following or a combination thereof : 

Reviewer: ```suggestion
[Cloud native](https://github.com/cncf/toc/blob/main/DEFINITION.md) project adopters may be any one of the following. Please note that any single company can fall under several categories at once. If that's the case, it should enumerate all that apply and only be listed once.
```

* A CNCF End-User member - Companies and organizations who are members of the CNCF (https://www.cncf.io/enduser/ & https://landscape.cncf.io/card-mode?enduser=yes)
* Another project - an open source project that leverages a CNCF project as part of their solution, integrates with for compatibility and interoperability, or is used in the supply chain of another project

> * A Service Provider - a service provider is a organization that repackages an open source project as a core component of a service offering, sells cloud native services externally

Reviewer: ```suggestion
* A Service Provider - a Service Provider is a organization that repackages an Open Source project as a core component of a service offering, sells Cloud Native services externally
```

Reviewer: I am unsure on capitalizing Cloud Native.

Author: i leverage https://github.com/cncf/toc/blob/main/DEFINITION.md for capitalization of cloud native.

Author: @RichiH  do we traditionally capitalize "open source"?

Reviewer 4: Not usually.

Reviewer: I thought we did; thanks for checking, closing.

  * APIs
  * SaaS
  * A service provider’s customers are considered transitive adopters and should be excluded from identification within the ADOPTERS.md file.

>   * Examples of service providers (and not end users) include cloud providers (e.g., Alibaba Cloud, AWS, Google Cloud, Microsoft Azure), consulting firms (e.g., Accenture, Booz Allen), infrastructure software vendors (e.g., SUSE, Red Hat), and telecom operators (e.g., AT&T, China Mobile). Refer to the [vendor category on the landscape](https://landscape.cncf.io/category=cncf-members&enduser=no&format=card-mode&grouping=category) for more examples.

Reviewer: ```suggestion
  * Examples of Service Providers (and not end users) include cloud providers (e.g., Alibaba Cloud, AWS, Google Cloud, Microsoft Azure), consulting firms (e.g., Accenture, Booz Allen), infrastructure software vendors (e.g., SUSE, Red Hat), and telecom operators (e.g., AT&T, China Mobile). Refer to the [vendor category on the landscape](https://landscape.cncf.io/category=cncf-members&enduser=no&format=card-mode&grouping=category) for more examples.
```

> * End users - companies and organizations that are not CNCF End-User members that use the project and cloud native technologies internally,  or build upon a cloud native open source project but do not sell the cloud native project externally as a service offering (those are service providers)

Reviewer: Should this be before Service Provider as that section references this section?

>   * Examples of Service Providers (and not end users) include cloud providers (e.g., Alibaba Cloud, AWS, Google Cloud, Microsoft Azure), consulting firms (e.g., Accenture, Booz Allen), infrastructure software vendors (e.g., SUSE, Red Hat), and telecom operators (e.g., AT&T, China Mobile). Refer to the [vendor category on the landscape](https://landscape.cncf.io/category=cncf-members&enduser=no&format=card-mode&grouping=category) for more examples.

Reviewer 3: ```suggestion
Examples of Service Providers (and not end-users) include cloud providers (e.g., Alibaba Cloud, AWS, Google Cloud, Microsoft Azure), consulting firms (e.g., Accenture, Booz Allen), infrastructure software vendors (e.g., SUSE, Red Hat), and telecom operators (e.g., AT&T, China Mobile). Refer to the [vendor category on the landscape](https://landscape.cncf.io/category=cncf-members&enduser=no&format=card-mode&grouping=category) for more examples.
```

> * Consultancy - an entity whose purpose is to assist other organizations in developing a solution leveraging cloud native technology. They may be embedded in the end user team and is responsible for the execution of the service. They may also package cloud native technologies for reuse as part of their offerings. These function as proxies for an end user.

Reviewer: This is now doubly defined, as it's also part of Service Provider.

Author: I think the language we used here needs more clarity to avoid the double definition.  Service Providers (listed above) explicitly offer as a service, whereas consultancies may offer packages of projects together as "their supply chain offering", but is not run as a service therefore they are not a Service Provider.  Suggestions?

Reviewer 2: Speaking as someone who ran a consulting company for 10 years, there is no such distinction in actual practice.

Reviewer: From my PoV: A Service Provider provides services under their own name. A consultancy provides knowledge and/or services which are offered under the name of the End User and/or Service Provider.

Reviewer 2: That definition of "consultancy" would include Accenture and Booz Allen.

>   * For the purposes of identifying adopters of a project, consultancies are not recommended for inclusion in the ADOPTERS.md file

Reviewer: ```suggestion
  * For the purposes of identifying adopters of a project, consultancies are not recommended for inclusion in the ADOPTERS.md file. They are, however, encouraged to confirm other End Users and/or Service Providers where possible.
```

> * Consultancy - an entity whose purpose is to assist other organizations in developing a solution leveraging cloud native technology. They may be embedded in the end user team and is responsible for the execution of the service. They may also package cloud native technologies for reuse as part of their offerings. These function as proxies for an end user.

Reviewer 2: Here you have consultancy as a category, but above you include Consulting Firms in Service providers.  What's the difference between a consultancy and a consulting firm?

Author: Thanks Josh for calling this out, @RichiH did as well and i missed removing this portion of the reference

Projects may leverage the above guidelines to list organizations in their ADOPTERS.md file within their repo.

> If you’re not sure if your organization falls into any of these categories you can email info@cncf.io and we’ll help you.

Reviewer: Should this go to TOC, instead?

Author: The original pointed to info@cncf.io.  I've added the #toc mailing list as well as the slack channel

>   * For the purposes of identifying adopters of a project, consultancies are not recommended for inclusion in the ADOPTERS.md file. They are, however, encouraged to confirm other End Users and/orService Providers where possible.

Reviewer 3: ```suggestion
To identify adopters of a project, consultancies are not recommended for inclusion in the ADOPTERS.md file. They are, however, encouraged to confirm other End Users and/or Service Providers where possible.
```

>   * For the purposes of identifying adopters of a project, consultancies are not recommended for inclusion in the ADOPTERS.md file. They are, however, encouraged to confirm other End Users and/orService Providers where possible.

Reviewer 5: Would it be possible to explain this decision a bit further on why consultancies are excluded?

I would have expected that would be helpful to know that a shop who may have X number of clients is developing solutions that use the project in question. While any particular clients may not be large enough to be interesting, if the consultant is using the project for a sufficient number of clients, that would be meaningful.

Reviewer 6: Consultancies are often in the role of "trusted expert advisors" who may recommend certain projects to their clients. If a consultancy is prepared to publicly call themselves a project adopter, that does seem like a meaningful vote of confidence

Reviewer 2: The one caution here is that consultancies often "support" particular technologies, as a marketing thing, despite having no actual clients that run them. But ... the same could be said of service providers.  I think what's appropriate is just that the sponsor needs to actually check that the software is, in fact, being used.

* End users - companies and organizations that are not CNCF End-User members that use the project and cloud native technologies internally,  or build upon a cloud native open source project but do not sell the cloud native project externally as a service offering (those are Service Providers). This group is identified in the written form by the convention `end user`, uncapitalized and unhyphenated.
* A Service Provider - a Service Provider is a organization that repackages an open source project as a core component of a service offering, sells cloud native services externally
  * APIs
  * SaaS
  * A Service Provider’s customers are considered transitive adopters and should be excluded from identification within the ADOPTERS.md file.
  * Examples of Service Providers (and not end users) include cloud providers (e.g., Alibaba Cloud, AWS, Google Cloud, Microsoft Azure), some infrastructure software vendors, and telecom operators (e.g., AT&T, China Mobile). Refer to the [vendor category on the landscape](https://landscape.cncf.io/category=cncf-members&enduser=no&format=card-mode&grouping=category) for more examples.

> * Consultancy - an entity whose purpose is to assist other organizations in developing a solution leveraging cloud native technology. They may be embedded in the end user team and is responsible for the execution of the service. Service Providers may also provide consultancy services, they may also package cloud native technologies for reuse as part of their offerings. These function as proxies for an end user.

Reviewer 2: +1 on this new language, thanks.
